src/WebJobs.Script.WebHost/WebHooks/DefaultScriptWebHookProvider.cs

```diff
@@ -33,14 +34,16 @@ public bool TryGetHandler(string name, out HttpHandler handler)
         // Exposed to extensions to get the URL for their http handler.
         public Uri GetUrl(IExtensionConfigProvider extension)
         {
-            var extensionType = extension.GetType();
             var handler = extension as HttpHandler;
             if (handler == null)
             {
                 throw new InvalidOperationException($"Extension must implement IAsyncConverter<HttpRequestMessage, HttpResponseMessage> in order to receive webhooks");
             }
 
-            string name = extensionType.Name;
+            // use the config section moniker for the extension as the URL name
+            var extensionType = extension.GetType();
+            var attrib = extensionType.GetCustomAttribute<ExtensionAttribute>();
+            string name = (attrib?.ConfigurationSection ?? extensionType.Name).ToLowerInvariant();
```

Comment:
ExtensionAttribute.Name isn't a programmatic moniker - it can contain spaces (human readable). So I use the config section name.